test/cpp/tensorexpr/test_simplify.cpp

```diff
@@ -3964,5 +3964,30 @@ void testSimplifyRampSubBroadcast() {
   ASSERT_EQ(newRamp->lanes(), num_lanes);
 }
 
+void testSimplifyBroadcastTermExpander() {
+  KernelScope kernel_scope;
+  int num_lanes = 8;
+  ExprHandle bc0 = Broadcast::make(ExprHandle(0), num_lanes);
+  ExprHandle bc1 = Broadcast::make(ExprHandle(1), num_lanes);
+  ExprHandle bc2 = Broadcast::make(ExprHandle(2), num_lanes);
+  // NB: We need a term in the middle which isn't simplified to trigger the
+  // relevant path in TermExpander::mutate.
+  ExprHandle simplified = IRSimplifier::simplify(bc1 + (bc0 / bc2) + bc1);
```

Comment:
I don't see how this tests the new code, theres no lane mismatch here is there?

Reply:
There is, without this change the test fails. The two `bc1` are brought together rewritten to `2 * bc1`, so a broadcast of 2 needs to be inserted. Without this change, we inserted a cast. This is the minimal example I could find triggering the problem, since it needs at least three terms AFAICT.

Reply:
Oh interesting, makes sense. I didn't test all combinations with lanes, so I'm quite happy the fix was this well contained.